base/sysimg.jl

```diff
@@ -3,18 +3,43 @@
 baremodule Base
 
 using Core.Intrinsics
-ccall(:jl_set_istopmod, Void, (Bool,), true)
-function include(path::AbstractString)
+ccall(:jl_set_istopmod, Void, (Any, Bool), Base, true)
+function include(mod::Module, path::AbstractString)
     local result
     if INCLUDE_STATE === 1
-        result = Core.include(path)
+        result = Core.include(mod, path)
     elseif INCLUDE_STATE === 2
-        result = _include(path)
+        result = _include(mod, path)
     elseif INCLUDE_STATE === 3
-        result = include_from_node1(path)
+        result = include_from_node1(mod, path)
+    end
+    result
+end
+function include(path::AbstractString)
+    local result
+    if INCLUDE_STATE === 1
+        result = Core.include(Base, path)
+    elseif INCLUDE_STATE === 2
+        result = _include(Base, path)
+    else
+        # to help users avoid error (accidentally evaluating into Base), this is deprecated
```

Comment:
if it should become an error eventually, then add a reminder note in deprecated.jl